### ISSUE TITLE
Adding/Removing mention while editing message does not affect Activity center notification

### ISSUE THREAD

**Issue by pavloburykh** [status-im/status-react#21284]
**Steps:** 

1. User A sends message in community channel
2. User A edits message by mentioning @User B 
3. See if User B receives notification in AC

**Actual result:** UserB does not receive notification in AC despite he was mentioned. Same works in opposite way: removing mention from initial message does not result in AC notification disappear for User who has been previously mentioned. 

https://github.com/user-attachments/assets/1084d4db-169c-4d80-8dd3-b8eb8e730a9a

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

**Comment by Horupa-Olena** [comment]
Still reproduce!

**Comment by ulisesmac** [comment]
AFAIK, the activity center state is received from status-go, I'm able to reproduce this same issue on Desktop too, so I think the root cause comes from `status-go`.

@ilmotta @flexsurfer does it make sense? wdyt?

**Comment by ilmotta** [comment]
@ulisesmac, even if it's caused by a bug in status-go, we should ideally fix it ourselves. Another option if you identified the bug also exists in Desktop is to see if `status-desktop` already has this same bug reported. Maybe a dev from Desktop can more quickly fix the bug.

**Comment by ulisesmac** [comment]
@ilmotta 

Yeah, I agree that we should fix it, but I wanted to know your opinion about my understanding of the problem, does it make sense that the root cause comes from status-go? I'm not too familiar with the code for the activity center and there are many signals happening making it hard to debug

**Comment by ilmotta** [comment]
> [@ilmotta](https://github.com/ilmotta)
> 
> Yeah, I agree that we should fix it, but I wanted to know your opinion about my understanding of the problem, does it make sense that the root cause comes from status-go? I'm not too familiar with the code for the activity center and there are many signals happening making it hard to debug

@ulisesmac, the way the AC was coded doesn't have a lot of surface for bugs because the logic controlling when signals are sent, how notifications are synced, all those important bits are handled by status-go. In the client we mostly fetch synchronously once (first page of the notifications) and then keep the app-db up-to-date based on the signals. So yes to your question, the reason is most likely in status-go, but I'd first check if the message sent via a signal from status-go is correct or not. If it's incorrect after editing or no message was sent at all, then the problem is most likely in status-go.

> there are many signals happening making it hard to debug

One thing I did a few times which might help you is to disable certain logs to focus only in one type.

Disable these:
https://github.com/status-im/status-mobile/blob/9c09fc709992801c04a1b047ee9564224f5f877e/src/status_im/common/signals/events.cljs#L34-L35

https://github.com/status-im/status-mobile/blob/9c09fc709992801c04a1b047ee9564224f5f877e/src/status_im/common/signals/events.cljs#L111

Then log at this level the message payload:
https://github.com/status-im/status-mobile/blob/9c09fc709992801c04a1b047ee9564224f5f877e/src/status_im/contexts/chat/messenger/messages/transport/events.cljs#L76

Or log directly the payload in the event `:activity-center.notifications/reconcile`

Another option is to simulate the bug reported in this issue as a unit test in status-go.

**Comment by churik** [comment]
@ilmotta I suggest to postpone it for now from 2.31, please move it back if you see that it is required

**Comment by churik** [comment]
Decided with @ilmotta to exclude it for now from 2.32 given the complexity
